### PR TITLE
Setup test vars

### PR DESCRIPTION
**Pull request by spetrosi**
When including the ha_cluster role in other roles, it is required to configure CI for ha_cluster to run smoothly in tests. Currently, i need to use relative path like this:
```
     - name: Set up test environment for the ha_cluster role
      include_role:
        name: fedora.linux_system_roles.ha_cluster
        tasks_from: ../../../tests/ha_cluster/tasks/setup_test.yml 
```
The relative path is weird because collections tests directory is far away from tasks directory.

In this PR I implement `__ha_cluster_setup_test` and `__ha_cluster_setup_sbd` variables to enhance this.

### CODE REVIEW

**Comment by richm** [comment]
[citest commit:ca2eb8923c89585bc579d7a69c9ee8e1948617df]

File: tasks/main.yml [hunk 1]
```diff
@@ -3,6 +3,14 @@
 - name: Set platform/version specific variables
   include_tasks: set_vars.yml
 
+- name: Set up test environment
+  include_tasks: setup_test.yml
+  when: __ha_cluster_setup_test | d(false)
+
+- name: Set up test environment for SBD
+  include_tasks: tasks/setup_sbd.yml
```
Comment (tomjelinek) [inline tasks/main.yml:11]:
```suggestion
  include_tasks: setup_sbd.yml
```
Like this, perhaps?

**Review by tomjelinek — CHANGES_REQUESTED** [review]
Generally, I don't like mixing production code and test code. But if there's no other way, then at least the test code should be clearly separated from production code. The `setup_sbd.yml` file could easily be considered a production code. In order to keep test code separated from production, I suggest the following renaming:
* `__ha_cluster_setup_test` -> `__ha_cluster_test_setup`
* `__ha_cluster_setup_sbd` -> `__ha_cluster_test_setup_sbd`
* `setup_test.yml` -> `test_setup.yml`
* `setup_sbd.yml` -> `test_setup_sbd.yml`

I appreciate the variables being prefixed with double underscore and not being defined in `defaults/main.yml`. This will hopefully discourage people from setting them.

Tests which run `setup_sbd.yml` also run `cleanup_sbd.yml`. Yet you haven't move this file. How do you do SBD cleanup in your tests?

I'm using `yes` and `no` in this role instead of `true` and `false`, respectively. Can you stick to that?

Thanks.

File: tasks/main.yml [hunk 2]
```diff
@@ -3,6 +3,14 @@
 - name: Set platform/version specific variables
   include_tasks: set_vars.yml
 
+- name: Set up test environment
+  include_tasks: setup_test.yml
+  when: __ha_cluster_setup_test | d(false)
```
[the remark this answers is not in the document]
Comment (richm) [reply tasks/main.yml:8]: > What does `| d` mean?

`d` is an alias for `default` https://jinja.palletsprojects.com/en/2.9.x/templates/#default

It's pretty common in other system roles to use `d`, especially to reduce verbosity in long jinja expressions

> Can it be `| default` instead?

Yes.

[the remark this answers is not in the document]
Comment (tomjelinek) [reply tasks/main.yml:8]: That's cool! I didn't know about this.

> It's pretty common in other system roles to use d, especially to reduce verbosity in long jinja expressions

I can see why :slightly_smiling_face:

**Comment by tomjelinek** [comment]
> The other way to do it is this https://github.com/linux-system-roles/mssql/pull/62/files#r875224828 which isn't very elegant.

I saw that and I agree that it is a worse solution that what is proposed in this PR.

> Suggestions for a better way to do this are welcome.

I was thinking about symlinking the `setup_test.yml` and `setup_sbd.yml` files either to a more suitable location in the `ha_cluster` role or directly to the `mssql` role. But I wasn't sure if this would work and could be packaged. And it still may brake if I decide to change these files. So I didn't even propose it.

Another possibility would be to extract these files into a separate repository / role. That doesn't seem to be worth the extra work and complications, though.

> Yes, but keep in mind that the recommendation is to use `true` and `false`.

Oh, ok. Somehow, I thought it was recommended to use `yes` and `no` instead. I'm fine with keeping `true`, `false` and `d` in place and I will update the rest of the role to use `true`, `false` and `d` to keep it consistent and go the recommended way.

**Comment by richm** [comment]
> > The other way to do it is this https://github.com/linux-system-roles/mssql/pull/62/files#r875224828 which isn't very elegant.
> 
> I saw that and I agree that it is a worse solution that what is proposed in this PR.
> 
> > Suggestions for a better way to do this are welcome.
> 
> I was thinking about symlinking the `setup_test.yml` and `setup_sbd.yml` files either to a more suitable location in the `ha_cluster` role or directly to the `mssql` role. But I wasn't sure if this would work and could be packaged. And it still may brake if I decide to change these files. So I didn't even propose it.
> 
> Another possibility would be to extract these files into a separate repository / role. That doesn't seem to be worth the extra work and complications, though.
> 
> > Yes, but keep in mind that the recommendation is to use `true` and `false`.
> 
> Oh, ok. Somehow, I thought it was recommended to use `yes` and `no` instead. I'm fine with keeping `true`, `false` and `d` in place and I will update the rest of the role to use `true`, `false` and `d` to keep it consistent and go the recommended way.

Ok, but I would consider that a low priority - at least for now, ansible-lint and ansible-test allow them.

**Comment by spetrosi** [comment]
\> Tests which run `setup_sbd.yml` also run `cleanup_sbd.yml`. Yet you haven't move this file. How do you do SBD cleanup in your tests?
I did the same thing with cleanup_sbd.yml for consistency, let me know what you think

**Review by tomjelinek — CHANGES_REQUESTED** [review]
`tests_sbd_all_options.yml` is failing. Previously, the test ran `
tests/tasks/setup_sbd.yml` first, which created block devices for SBD. Then the test ran `Set SBD devices and watchdogs` task, which set input variables for the role using a variable defined by `
tests/tasks/setup_sbd.yml`. Then it ran the role.

With this pull request, `setup_sbd.yml` is run as a part of the role and that cannot possibly work. The block devices (test's prerequisites) must be created firts and passed to the role, they cannot be created by the role itself.

`tests_cluster_advanced_knet_full.yml` fails for the same reason: setup, which the test depends on to set input variables for the role, got moved into the role.

Could your way of running `cleanup_sbd.yml` be used for the other two taskfiles as well, instead of passing variables to the role to run them?

**Comment by tomjelinek** [comment]
rhel-8 failures are expected, they are caused by not using the latest pcs version. Fixed pcs versions are pcs-0.10.10-4.el8_5.1 and pcs-0.10.12-3.el8

I'm not sure what's going on with Fedora 35, the ansible log is empty.

**Comment by richm** [comment]
> rhel-8 failures are expected, they are caused by not using the latest pcs version. Fixed pcs versions are pcs-0.10.10-4.el8_5.1 and pcs-0.10.12-3.el8
> 
> I'm not sure what's going on with Fedora 35, the ansible log is empty.

Fedora stopped publishing a latest F35 image :-( https://pagure.io/cloud-sig/issue/379 - I don't suppose you know someone working on the Fedora Cloud SIG team . . .

**Comment by richm** [comment]
one minor nit, otherwise, lgtm

File: tasks/test_cleanup_sbd.yml [hunk 3]
```diff
@@ -0,0 +1,17 @@
+# SPDX-License-Identifier: MIT
+# This task configures test systems for testing in the CI environment.
+# Only use this task for testing.
+# This task file resides in `tasks/` so that it can be used from other roles
+# with `include_role` with `tasks_from: filename` directly without the need to
+# provide a relative path. Providing relative path is problematic with
+# collections.
+---
+- name: Unount SBD devices
```
[the remark this answers is not in the document]
Comment (spetrosi) [reply tasks/test_cleanup_sbd.yml]: Fixed

**Comment by tomjelinek** [comment]
> Fedora stopped publishing a latest F35 image :-( https://pagure.io/cloud-sig/issue/379

Thanks for the info.

> I don't suppose you know someone working on the Fedora Cloud SIG team . . .

I don't.

**Pull request merged**